src/api/base/login.ts

```diff
@@ -16,38 +16,37 @@ const getSoapBody = (props: Credentials) => {
 </soapenv:Envelope>`
 }
 
-export const login = (props: Credentials): Promise<BlueAntSession> => {
-  return new Promise((resolve, reject) => {
+export const login = async (props: Credentials): Promise<BlueAntSession> => {
+  return await new Promise((resolve, reject) => {
     axios
       .post(
         `/services/${import.meta.env.VITE_API_SERVICE_BASE}`,
         getSoapBody(props),
-        { headers: getHeaders("Login") }
+        { headers: getHeaders('Login') }
       )
       .then((response) => {
         const parsed = new XMLParser().parse(response.data)
         const session =
-          parsed["soapenv:Envelope"]["soapenv:Body"]["ns2:session"]
+          parsed['soapenv:Envelope']['soapenv:Body']['ns2:session']
         resolve({
-          personID: session["ns2:personID"],
-          sessionID: session["ns2:sessionID"],
+          personID: session['ns2:personID'],
+          sessionID: session['ns2:sessionID']
         })
       })
       .catch((error) => {
-        let responseMessage: string | undefined = undefined
+        let responseMessage: string | undefined
 
-        if (error.data) {
+        if (error.data !== undefined) {
```

Comment:
lodash isNil

Reply:
Do not
import * as _ from 'lodash'
import {isNil} from 'lodash

Do
import isNil from 'lodash/isNil'

Reply:
Done